src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -452,8 +447,11 @@ public function nack(\AMQPEnvelope $message, string $queueName, int $flags = \AM
 
     public function setup(): void
     {
-        $this->setupExchangeAndQueues();
+        if ($this->autoSetupExchange) {
```

Comment:
The setup seems broken now. The setup method is public because it's called by SetupableTransportInterface. Now if you disable auto_setup and want to setup the transports manually, then it does not actually setup the exchanges and queues anymore. But that is the whole point of the setup method: being able to setup manually when autosetup is not used.

Reply:
Fixed in https://github.com/symfony/symfony/pull/40966